pkg/domain/resourcegroup/runaway.go

```diff
@@ -496,9 +500,16 @@ func (r *RunawayChecker) BeforeCopRequest(req *tikvrpc.Request) error {
 	if r.setting == nil {
 		return nil
 	}
-	marked := r.marked.Load()
+	marked := r.markedByRule.Load()
```

Comment:
```suggestion
	marked := r.markedByWatch || r.markedByRule.Load() 
```

Reply:
If the action in group setting is kill, but the action in the query watch setting is cooldown, and markedByRule is false, we should use the action in the query watch setting